google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -113,12 +113,18 @@ public CancellableRunnable createDeployRunner(
       AppEngineDeploymentConfiguration deploymentConfiguration,
       DeploymentOperationCallback callback) {
 
-    if (source.getFile() == null
-        || !(source instanceof AppEngineDeployable)) {
-      callback.errorOccurred(GctBundle.message("appengine.deployment.source.not.found.error"));
+    if (!(source instanceof AppEngineDeployable)) {
+      callback.errorOccurred(GctBundle.message("appengine.deployment.invalid.source.error"));
       throw new RuntimeException("Invalid deployment source selected for deployment");
     }
 
+    if (source.getFile() == null
+        || !source.getFile().exists()) {
+      callback.errorOccurred(GctBundle.message("appengine.deployment.source.not.found.error",
+          source.getFilePath()));
+      throw new RuntimeException("Deployment source selected for deployment not found");
```

Comment:
On second thought, I think its ok to keep the exception here: if the DeploymentSource was created in the first place then there should be a valid path to a deployable file (or exploded war directory). It is an exceptional case that this will not happen.

What do you think?